### ISSUE TITLE
Memory leakage

### ISSUE THREAD

**Issue by DingMaolinBio** [JackieHanLab/TOSICA#16]
Hi, I am sorry to bother you. When I tried to apply the TOSICA, I was confused in installing the environment. Here are details:
1. When I used packages (pytorch=1.7.1 torchvision=0.8.2 torchaudio=0.7.2 cudatoolkit=10.1), the training process would stucked while move the model to the GPU;
2. When I used the environment that included packages(pytorch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 cudatoolkit=11.6), the   consumption of the GPU would increase while training. I guess it could be the leakage of the memory.

Could you offer us some suggestions about these situation? Thanks for your reply.

**Comment by GlancerZ** [comment]
> Hi, I am sorry to bother you. When I tried to apply the TOSICA, I was confused in installing the environment. Here are details:
> 
> 1. When I used packages (pytorch=1.7.1 torchvision=0.8.2 torchaudio=0.7.2 cudatoolkit=10.1), the training process would stucked while move the model to the GPU;
> 2. When I used the environment that included packages(pytorch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 cudatoolkit=11.6), the   consumption of the GPU would increase while training. I guess it could be the leakage of the memory.
> 
> Could you offer us some suggestions about these situation? Thanks for your reply.

Do you have any idea, I also meet this problem.

**Comment by DingMaolinBio** [comment]
Thanks for your request. But I am sorry to tell you that I have not make any progress in handling this issue. At last, I tried to train the model with CPU, but it took too much time. I am truly sorry.发自我的 iPhone在 2023年9月23日，12:45，GlancerZ ***@***.***> 写道：﻿

Hi, I am sorry to bother you. When I tried to apply the TOSICA, I was confused in installing the environment. Here are details:

When I used packages (pytorch=1.7.1 torchvision=0.8.2 torchaudio=0.7.2 cudatoolkit=10.1), the training process would stucked while move the model to the GPU;
When I used the environment that included packages(pytorch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 cudatoolkit=11.6), the   consumption of the GPU would increase while training. I guess it could be the leakage of the memory.

Could you offer us some suggestions about these situation? Thanks for your reply.

Do you have any idea, I also meet this problem.

**Comment by JiaweiChenGo** [comment]
Thank you for your interest in TOSICA.
Thanks to @sldyns, we have resolved the issue of continuous GPU memory growth by altering weights = attn to weights = attn.detach() in TOSICA_model.py.
